### DISCUSSION TITLE
lelist&telist

### DISCUSSION THREAD

**Discussion by byeonghoon-ahn** [mdolab/dafoam#184] [General]
Hello, I'm learning the example of the wing. In the previous examples and some descriptions, 
both lelist and telist should be in the surface, and the height in Z direction at the wing tip in FFD,
 but tip in the code z in tip_pt and break_pt are set to 0, so it can't be generated in the surface?, I don't understand much about these codes

LE_pt = np.array([0.01, 0.01, 0.0])
break_pt = np.array([0.848, 1.119, 0.0])
tip_pt = np.array([2.855, 3.755, 0.0])

**Comment by friedenhe** [comment]
Can you be more specific? which case this is?

**Reply by byeonghoon-ahn** [reply]
Ok, this case is CRM wing tutorial, As i know lelist and telist shuold be close to the leading edge and completely within the wing surface mesh for thickness and volume constraints.  I can find wing tip(tip_pt) and break_pt doesn't locate in z =0, only wing root get z=0 in under FFD fig, but the code of break_pt and tip_pt's z direction are still 0 in runScript file,  why does z =0 of wing_tip and break_pt?, How they satisfy volume constraints and thickness constraints ?

![crm wing](https://user-images.githubusercontent.com/77006538/138193513-8d61b9bc-c231-4b84-b53c-1a8709788797.jpeg)

**Reply by friedenhe** [reply]
The leList and teList have to be completely within the wing planform. It will project up and down (in the z-direction ) to the wing surface. So the z coordinates do not matter that much. In this case, z=0 for leList and teList, so when they are projected up, it will get two intersections. pyGeo will sort these two and get the actual up and down projected points. Check this https://github.com/mdolab/pygeo/blob/bfdf1b58579f880afadcbfb9a54fdd515b6d9281/pygeo/constraints/DVCon.py#L2984

**Reply by byeonghoon-ahn** [reply]
Thanks a lot!
